### PR TITLE
Homework 3 done!

### PR DESCRIPTION
**Pull request by stekkko**
Some usefull links:
[Homework condition](http://www.seas.upenn.edu/~cis194/spring13/hw/03-rec-poly.pdf)
[Homeworks](http://www.seas.upenn.edu/~cis194/spring13/lectures.html)

Waiting for [Dmitriy](https://github.com/ChShersh) to review my shit code :D

### CODE REVIEW

File: src/Golf.hs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+skip (x:xs) 0 b = x : skip xs b b 
+skip (_:xs) a b = skip xs (a - 1) b
+
+skipskip :: [a] -> Int -> [[a]]
+skipskip list 0 = [list]
+skipskip list x = skipskip list (x - 1) ++ [skip list x x] 
+
+skips :: [a] -> [[a]]
+skips []   = []
+skips list = skipskip list (length list - 1)
```
Comment (chshersh) [inline src/Golf.hs]: Try to use more standard functions.

1. Write function `skipNth` which takes number `n` and remove every `n`th number from list. I see that your `skip` function does something similar. But it's more efficient to use standard functions. For `skipNth` you should use standard `splitAt` function:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Prelude.html#v:splitAt

2. Then You can apply `skipNth` function for every integer. Use `map` function for this. And probably something else.

File: src/Golf.hs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+--exercize 2--
+localMaxima :: [Integer] -> [Integer]
+localMaxima (x:y:z:xs) = [y | y > x && y > z] ++ localMaxima (y:z:xs)
```
Comment (chshersh) [inline src/Golf.hs]: This is a good solution! But you can use as-patterns to make it more ~scary~ shorter. Like this:

```haskell
localMaxima (x:xs@(y:z:_)) = [y | y > x && y > z] ++ localMaxima xs
```

File: src/Golf.hs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+localMaxima _          = []
+
+--exercize 3--
+count :: [Integer] -> Integer -> Integer -> Integer
+                    -> Integer -> Integer -> Integer
+                     -> Integer -> Integer -> Integer
+                      -> Integer -> [Integer]
+count (x:xs) a0 a1 a2 a3 a4 a5 a6 a7 a8 a9 = case x of
```
Comment (chshersh) [inline src/Golf.hs]: This function is really ugly 😕 You shouldn't write such functions! Check documentation for `Data.List` module and see what you can use to make cleaner code.

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-List.html

Probably you can find function like `transpose` useful:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-List.html#v:transpose

File: src/Golf.hs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                                         8 -> count xs a0 a1 a2 a3 a4 a5 a6 a7 (a8 + 1) a9
+                                         9 -> count xs a0 a1 a2 a3 a4 a5 a6 a7 a8 (a9 + 1)
+                                         _ -> count xs a0 a1 a2 a3 a4 a5 a6 a7 a8 a9
+count _ a0 a1 a2 a3 a4 a5 a6 a7 a8 a9 = [a0, a1, a2, a3, a4, a5, a6, a7, a8, a9]
+
+putLn :: [Integer] -> Integer -> String
+putLn [] _ = []
+putLn (x:xs) m = (if x == m then '*' else ' ') : putLn xs m
+
+decMaxByOne :: [Integer] -> Integer -> [Integer]
```
Comment (chshersh) [inline src/Golf.hs]: `Integer` is not very efficient type. If you know that your input won't too big, it's better to use more compact types like `Int`.

**Review by chshersh — CHANGES_REQUESTED** [review]

File: src/Golf.hs (hunk 4 above)
Comment (stekkko) [reply src/Golf.hs]: I just always use Integer because I don't want to see exceptions that Integer doesn't match Int or something like that.

Comment (chshersh) [reply src/Golf.hs]: It's okay for homeworks but in real life you should use efficient types if possible.

**Review by chshersh — APPROVED** [review]
Great! Some improvements can be made. But still quite good :) :+1:

**Pull request merged**